plugins/extractors/metabase/extractor.go

```diff
@@ -0,0 +1,213 @@
+package metabase
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/odpf/meteor/core"
+	"github.com/odpf/meteor/core/extractor"
+	"github.com/odpf/meteor/plugins"
+	meteorMeta "github.com/odpf/meteor/proto/odpf/meta"
+	"github.com/odpf/meteor/utils"
+)
+
+var (
+	client = &http.Client{
+		Timeout: 2 * time.Second,
+	}
+	session_id = ""
+)
+
+type Config struct {
+	UserID    string `mapstructure:"user_id" validate:"required"`
+	Password  string `mapstructure:"password" validate:"required"`
+	Host      string `mapstructure:"host" validate:"required"`
+	SessionID string `mapstructure:"session_id"`
+}
+
+type responseID struct {
+	ID string `json:"id"`
+}
+
+type Dashboard struct {
+	ID          int `json:"id"`
+	Urn         string
+	Name        string  `json:"name"`
+	Source      string  `default:"metabase"`
+	Description string  `json:"description"`
+	Charts      []Chart `json:"ordered_cards"`
+}
+
+type Chart struct {
+	ID           int `json:"card_id"`
+	Urn          string
+	Source       string `default:"metabase"`
+	DashboardUrn string
+	DashboardID  int `json:"dashboard_id"`
+}
+
+type Extractor struct {
+	out chan<- interface{}
+
+	logger plugins.Logger
+}
+
+func New(logger plugins.Logger) *Extractor {
+	return &Extractor{
+		logger: logger,
+	}
+}
+
+// Extract collects metdata from the source. Metadata is collected through the out channel
+func (e *Extractor) Extract(ctx context.Context, config map[string]interface{}, out chan<- interface{}) (err error) {
+	e.out = out
+	// Build and validate config received from receipe
+	var cfg Config
+	if err := utils.BuildConfig(config, &cfg); err != nil {
+		return extractor.InvalidConfigError{}
+	}
+	// get session id for further api calls in metabase
+	if cfg.SessionID == "" {
+		err = e.getSessionID(cfg)
+		if err != nil {
+			return
+		}
+	} else {
+		session_id = cfg.SessionID
+	}
+	var dashboards []Dashboard
+	err = e.getDashboardsList(cfg, &dashboards)
+	if err != nil {
+		return
+	}
+	for _, dashboard := range dashboards {
+		dashboardInfo, err := e.getDashboardInfo(strconv.Itoa(dashboard.ID), cfg, dashboard.Name)
+		if err != nil {
+			return err
+		}
+		Charts := make([]*meteorMeta.Chart, len(dashboardInfo.Charts))
+		for i, chart := range dashboardInfo.Charts {
+			c := e.metabaseCardToMeteorChart(chart.Source, chart.DashboardUrn, chart.Urn)
+			Charts[i] = &c
+		}
+		e.out <- meteorMeta.Dashboard{
```

Comment:
since we use the `out` channel here, do we still need to store it as a private field?